src/DataGrid.tsx

```diff
@@ -215,96 +267,214 @@ function DataGrid<R, K extends keyof R, SR>({
     };
   }, [width]);
 
-  function handleColumnResize(column: CalculatedColumn<R, SR>, width: number) {
+  useEffect(() => {
+    if (!onSelectedRowsChange) return;
+
+    const handleRowSelectionChange = ({ rowIdx, checked, isShiftClick }: SelectRowEvent) => {
+      assertIsValidKey(rowKey);
+      const newSelectedRows = new Set(selectedRows);
+      const rowId = rows[rowIdx][rowKey];
+
+      if (checked) {
+        newSelectedRows.add(rowId);
+        const previousRowIdx = lastSelectedRowIdx.current;
+        lastSelectedRowIdx.current = rowIdx;
+        if (isShiftClick && previousRowIdx !== -1 && previousRowIdx !== rowIdx) {
+          const step = Math.sign(rowIdx - previousRowIdx);
+          for (let i = previousRowIdx + step; i !== rowIdx; i += step) {
+            newSelectedRows.add(rows[i][rowKey]);
+          }
+        }
+      } else {
+        newSelectedRows.delete(rowId);
+        lastSelectedRowIdx.current = -1;
+      }
+
+      onSelectedRowsChange(newSelectedRows);
+    };
+
+    return eventBus.subscribe('SELECT_ROW', handleRowSelectionChange);
+  }, [eventBus, onSelectedRowsChange, rows, rowKey, selectedRows]);
+
+  useImperativeHandle(ref, () => ({
+    scrollToColumn(idx: number) {
+      scrollToCell({ idx });
+    },
+    scrollToRow(rowIdx: number) {
+      const { current } = gridRef;
+      if (!current) return;
+      current.scrollTop = rowIdx * rowHeight;
+    },
+    selectCell(position: Position, openEditor?: boolean) {
+      eventBus.dispatch('SELECT_CELL', position, openEditor);
+    },
+    openCellEditor(rowIdx: number, idx: number) {
+      eventBus.dispatch('SELECT_CELL', { rowIdx, idx }, true);
+    }
```

Comment:
Since these functions are only used for the ref handle, I've inlined them inside the `useImperativeHandle` function call.
That way we don't create possibly unused functions.